source/server/config_validation/server.cc

```diff
@@ -59,7 +59,7 @@ ValidationInstance::ValidationInstance(
       mutex_tracer_(nullptr), grpc_context_(stats_store_.symbolTable()),
       http_context_(stats_store_.symbolTable()), router_context_(stats_store_.symbolTable()),
       time_system_(time_system), server_contexts_(*this),
-      quic_stat_names_(stats_store_.symbolTable()) {
+      quic_context_(stats_store_.symbolTable(), absl::nullopt) {
```

Comment:
I'm slightly surprised that this is a nullopt here. Does this maybe deserve a comment explaining why? I have a couple of hypothesis, but I'm not sure if they're valid :)

Reply:
I can explain why -- the validation-server does not have a hot-restart object. You could arguably instantiate a no-op hot-restarter like we do if we disable hot-restart, but this seems OK as long as we always check the optional value before dereferencig it.

Reply:
> I can explain why -- the validation-server does not have a hot-restart object. 

Ah! That makes sense.

> You could arguably instantiate a no-op hot-restarter like we do in server/server.cc if we disable hot-restart, but this seems OK as long as we always check the optional value before dereferencig it.

*nod* that makes sense. Though if we do have a no-op restarter, I would probably be inclined to use it here so we can avoid the need to do the checks elsewhere. But this current approach is also just fine.

Reply:
I went with OptRef rather than the no-op object because getting the no-op one into all the various test contexts where the object is referenced would be way more of a pain than the one-line "is it set" check at the single usage site. I agree with the design preference of using the no-op object, but from a practical perspective doing it that way is a lot. (And the usage site isn't in any main flow so it's not performance-critical enough to warrant that effort.)